go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -97,29 +99,39 @@ func newEgressFirewallRule(rawEgressFirewallRule egressfirewallapi.EgressFirewal
 
 // NOTE: Utilize the fact that we know that all egress firewall related setup must have a priority: types.MinimumReservedEgressFirewallPriority <= priority <= types.EgressFirewallStartPriority
 func (oc *Controller) syncEgressFirewall(egressFirewalls []interface{}) {
+	err := wait.PollImmediate(500*time.Millisecond, 60*time.Second, func() (bool, error) {
+		if err := oc.syncEgressFirewallRetriable(egressFirewalls); err != nil {
+			klog.Errorf("Failed (will retry) in egress firewall: %v", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		klog.Fatalf("Error in syncing egress firewall: %v", err)
+	}
+}
+
+func (oc *Controller) syncEgressFirewallRetriable(egressFirewalls []interface{}) error {
```

Comment:
a good opportunity to add docstring ?

Reply:
how about 
```
// This function implements the main body of work of what is described by syncEgressFirewall.
// Upon failure, it may be invoked multiple times in order to avoid a pod restart.
```